### ISSUE TITLE
Improve MucButton accessibility in disabled state

### ISSUE THREAD

**Issue by markostreich** [it-at-m/muc-patternlab-vue#319]
**Is your feature request related to a problem? Please describe.**
The MucButton component is currently not accessible when it is in a disabled state. Using the native disabled attribute prevents users from tabbing over the button, making it effectively invisible to keyboard navigation and screen readers.

**Describe the solution you'd like**
Replace the native disabled attribute with the aria-disabled attribute. Enhance the component's behavior using JavaScript and CSS to ensure it still prevents clicks while maintaining accessibility.
